### PR TITLE
Decrease debug level for boost package

### PR DESCRIPTION
**Pull request by Artalus**
Specify library name and version:  **Boost**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

There is this comment around line 434:
```
# -d2 is to print more debug info and avoid travis timing out without output
```
However the package is not built with Travis anymore, and if it is - I believe such option should go to `CONAN_OPTIONS` or like in `.travis.yml`. During our GHA runs, Boost builds (debug+release, static+shared) result in +-60 MB of logs per compiler, more than 400k lines long, which is... quite unreadable, esp. since it mostly consists of lines like
```
2020-05-12T08:37:55.2949892Z common.copy /home/conan/.conan/data/boost/1.69.0/_/_/package/28221a1f80ccfa6a971856cbd200b515160f78b1/include/boost/io_fwd.hpp
2020-05-12T08:37:55.2951451Z 
2020-05-12T08:37:55.2954182Z     cp "/home/conan/.conan/data/boost/1.69.0/_/_/source/boost_1_69_0/boost/io_fwd.hpp"  "/home/conan/.conan/data/boost/1.69.0/_/_/package/28221a1f80ccfa6a971856cbd200b515160f78b1/include/boost/io_fwd.hpp"
```

Without `-d2`, the logs become about 120k lines and 20 MB. I would also probably even got rid of those `common.copy` lines, since they take about 80% of the log, but this seems to be the default behavior.

### CODE REVIEW

**Comment by uilianries** [comment]
I don't like to change default options, but it will reflect only to output, it's okay for me.

@Artalus could you please fix those errors related to end-line? They are new because we added a new hook.

**Comment by SSE4** [comment]
@uilianries should be okay to change that particular option, as it doesn't affect on package id

**Comment by Artalus** [comment]
I made #1630 to fix endline errors, but @madebr pointed out there is #1612 already that fixes said files. And as SSE4 pointed out, the `debug_level` option does not affect the package id.

**Comment by Artalus** [comment]
The errors were fixed by merging #1412. Could someone please restart the build and check what is wrong with CLA?

**Comment by uilianries** [comment]
Sometimes CLA doesn't work

File: recipes/boost/all/conanfile.py [hunk 1]
```diff
@@ -383,7 +383,9 @@ def _build_bcp(self):
         folder = os.path.join(self.source_folder, self._folder_name, 'tools', 'bcp')
         with tools.vcvars(self.settings) if self._is_msvc else tools.no_op():
             with tools.chdir(folder):
-                command = "%s -j%s --abbreviate-paths -d2 toolset=%s" % (self._b2_exe, tools.cpu_count(), self._toolset)
+                command = "%s -j%s --abbreviate-paths toolset=%s" % (self._b2_exe, tools.cpu_count(), self._toolset)
+                if self.options.debug_level:
+                    command.append(" -d%d" % self.options.debug_level)
```
Comment (madebr) [inline recipes/boost/all/conanfile.py:388]:
```suggestion
                    command += " -d%d" % self.options.debug_level
```

File: recipes/boost/all/conanfile.py [hunk 2]
```diff
@@ -53,7 +53,7 @@ class BoostConan(ConanFile):
         "lzma": [True, False],
         "zstd": [True, False],
         "segmented_stacks": [True, False],
-        "debug_level": [i for i in range(1, 14)],
+        "debug_level": [i for i in range(0, 14)],
```
Comment (madebr) [inline recipes/boost/all/conanfile.py:56]: I don't think this option changes the generated package.
=> remove it in `package_id(self)`?

Comment (Artalus) [reply recipes/boost/all/conanfile.py:56]: It is already removed there :)

Comment (madebr) [reply recipes/boost/all/conanfile.py:56]: hehe, sorry.
I thought for a minute you were adding the option.

**Comment by danimtb** [comment]
> @uilianries should be okay to change that particular option, as it doesn't affect on package id

@SSE4 Are you sure it does not affect the package id? I think it is not deleted from the package_id in the recipe, or maybe I failed to find it 😕

**Comment by Artalus** [comment]
> or maybe I failed to find it

Line 164 in `def package_id` upstream https://github.com/conan-io/conan-center-index/blob/master/recipes/boost/all/conanfile.py#L164

**Comment by Artalus** [comment]
...on a second thought, I kinda missed that it is under `else` :thinking: 

Does `header_only()` call also prevents options from package id generation, or is this a bug in recipe?

**Pull request merged**